src/lib/features/project/project-store.ts

```diff
@@ -637,10 +637,17 @@ class ProjectStore implements IProjectStore {
             .joinRaw('CROSS JOIN total')
             .whereBetween('rank', [offset + 1, offset + limit]);
         const rows = await query;
-        const applications = this.getAggregatedApplicationsData(rows);
+        if (rows.length !== 0) {
+            const applications = this.getAggregatedApplicationsData(rows);
+            return {
+                applications,
+                total: Number(rows[0].total) || 0,
+            };
+        }
+
         return {
-            applications,
-            total: Number(rows[0].total) || 0,
+            applications: [],
+            total: 0,
```

Comment:
Interesting. Did this somehow not work before or are you just saving a DB call when we don't need it?

Reply:
If rows count is 0 it will fail, and making it more readable